distributed/scheduler.py

```diff
@@ -3422,9 +3422,20 @@ def heartbeat_worker(
         ws._last_seen = time()
 
         if executing is not None:
-            ws._executing = {
-                parent._tasks[key]: duration for key, duration in executing.items()
-            }
+            ws._executing = {}
+            for key, duration in executing.items():
+                ts = parent.tasks[key]
```

Comment:
The `_` is for typed attributes available from the `parent` class (`SchedulerState`). If we annotate `ts` as `TaskState`, Cython knows the type and will optimize attribute access, etc.

```suggestion
                ts: TaskState = parent._tasks[key]
```